### PR TITLE
Rebalance infra componentes if correct label is selected on extra machinepool

### PR DESCRIPTION
**Pull request by morenod**
If extra machinepool labels contains the one related to infra (node-role.kubernetes.io/infra=) on its list of labels, stateful set of prometheus will be rolled to be placed on the infra nodes.

I added a validation on the --machinepool-labels to validate its format before creating anything.

Valid values:
node-role.kubernetes.io/infra=
node-role.kubernetes.io/infra=,node-role.kubernetes.io/logging=

Invalid values:
node-role.kubernetes.io/infra
node-role.kubernetes.io/infra=,node-role.kubernetes.io/logging=

Basically, it must contains a separated-comma-list of valid label, considering a valid label a key=value format, where value can be null

### CODE REVIEW

**Review by krishvoor — APPROVED** [review]
lgtm

File: rosa-hypershift/rosa-hosted-wrapper.py [hunk 1]
```diff
@@ -620,6 +617,11 @@ def _build_cluster(ocm_cmnd, rosa_cmnd, cluster_name_seed, must_gather_all, prov
                         logging.error("Insufficient number of workers on extra machinepool %s (%d). Expected: %d. Disabling e2e-benchmarking execution on %s" % (args.machinepool_name, extra_pool_workers, args.machinepool_replicas, cluster_name))
                         cluster_load = False
                         metadata['status'] = "Ready. Not enough workers on extra pool"
+                    elif args.machinepool_labels and "node-role.kubernetes.io/infra" in args.machinepool_labels:
```
Comment (smalleni) [inline rosa-hypershift/rosa-hosted-wrapper.py:620]: Is this really an extra machinepool or just few nodes on the same HC nodepool with this nodeselector?

Comment (smalleni) [reply rosa-hypershift/rosa-hosted-wrapper.py:620]: @morenod ^
